interfaces/repository/project_impl.go

```diff
@@ -0,0 +1,128 @@
+package repository
+
+import (
+	"github.com/gofrs/uuid"
+	"github.com/traPtitech/traPortfolio/domain"
+	"github.com/traPtitech/traPortfolio/interfaces/database"
+	"github.com/traPtitech/traPortfolio/interfaces/repository/model"
+	"github.com/traPtitech/traPortfolio/usecases/repository"
+)
+
+type ProjectRepository struct {
+	h database.SQLHandler
+}
+
+func NewProjectRepository(sql database.SQLHandler) *ProjectRepository {
+	return &ProjectRepository{h: sql}
+}
+
+func (repo *ProjectRepository) GetProjects() ([]*domain.Project, error) {
+	projects := []*model.Project{}
+	err := repo.h.Find(&projects).Error()
+	if err != nil {
+		return nil, err
+	}
+	res := make([]*domain.Project, 0, len(projects))
+	for _, v := range projects {
+		p := &domain.Project{
+			ID:          v.ID,
+			Name:        v.Name,
+			Since:       v.Since,
+			Until:       v.Until,
+			Description: v.Description,
+			Link:        v.Link,
+			CreatedAt:   v.CreatedAt,
+			UpdatedAt:   v.CreatedAt,
+		}
+		res = append(res, p)
+	}
+	return res, nil
+}
+
+func (repo *ProjectRepository) GetProject(id uuid.UUID) (*domain.Project, error) {
+	project := &model.Project{ID: id}
+	if err := repo.h.First(&project).Error(); err != nil {
+		return nil, err
+	}
+	res := &domain.Project{
+		ID:          project.ID,
+		Name:        project.Name,
+		Since:       project.Since,
+		Until:       project.Until,
+		Description: project.Description,
+		Link:        project.Link,
+		CreatedAt:   project.CreatedAt,
+		UpdatedAt:   project.CreatedAt,
+	}
+	return res, nil
+}
+
+func (repo *ProjectRepository) GetProjectMembers(id uuid.UUID) ([]*domain.ProjectMember, error) {
+	members := make([]*domain.ProjectMember, 0)
+	selectQuery := "users.id as user_id, users.name, project_members.since, project_members.until"
+	whereQuery := "project_members.project_id = ?"
+	joinQuery := "left join users on users.id = project_members.user_id"
+	err := repo.h.Model(&model.ProjectMember{}).Select(selectQuery).Where(whereQuery, id).Joins(joinQuery).Scan(&members).Error() //TODO 2回目からエラーが出る
+	if err != nil {
+		return nil, err
+	}
+	return members, nil
+}
```

Comment:
ここで1回目は成功するのですが、2回目からは500が出てしまいます
`Error 1054: Unknown column 'project_members.since' in 'field list'`と出るのですが、1回目だけ成功するのもよくわかっていません
この後に`GET /users`とかを叩いても↑のエラーが出るのでsqlhandlerが問題を起こしてるのかと思いましたが、解決はしていません、、
もし解決策があれば教えて頂けると助かります:pray:

Reply:
`members`は`model.ProjectMember`のほうがいいのと、バッククォートで囲んだ方が間違いがないと思います
ただこれをやってもうまく行かなかったので他のどこかで間違っている気がしています

ログを有効にしてみた感じだと、`project_members`じゃなくて`project`からsinceを取得しようとしてエラーになってるように見えました
[https://github.com/traPtitech/traPortfolio/blob/9f7a11df45b2dd583aa3642db0b0736aa3affd2d/infrastructure/sqlhandler.go#L74](ここで)ログを有効化できるのでそれをみてみるとわかるかもしれないです

Reply:
`members`のやつはそんなことはないかもしれないです

Reply:
sqlhandlerが間違っていたので直しました。
直す前のやつのログを見てもらえればわかるんですが、`left join ~~`の当たりがリクエストするたびに増えていて（`left join ~~ left join ~~`)みたいになっていました。これは多分gormのjoinsで指定したものが残ってしまっていたためにおこっていて、それを防ぐために最初のまっさらなDBを`sqlhandler.orig`に持つようにして`Error()`呼んだときにそれに置き換えるようにしました。
gorm v2のフィニッシャーみたいな感じだと思います

Reply:
#61 で修正しました🙏